src/modules/sign/components/gitSignIn.jsx

```diff
@@ -0,0 +1,54 @@
+import React, { useEffect, useState} from "react";
+import github from "../../../../src/assets/svg/github.svg";
+import activateGithub from "../../../../src/assets/svg/githubActive.svg";
+import gitlab from "../../../../src/assets/svg/gitlab.svg";
+import activateGitlab from "../../../../src/assets/svg/gitlabActive.svg";
+
+function GitSignIn() {
+  const [githubActive, githubSetActive] = useState(false);
+  const [gitlabActive, gitlabSetActive] = useState(false);
+  const githubActiveStyles = githubActive
+    ? {
+      border: "1px solid black",
+    }
+    : {};
+  const gitlabActiveStyles = gitlabActive
+    ? {
+      border: "1px solid black",
+    }
+    : {};
+  useEffect((e) => {
+    console.log(e);
+    const githubButton = document.getElementById("button");
+    githubButton.addEventListener("mouseover", function github () { githubSetActive(true) });
+    githubButton.addEventListener("mouseout", function github () { githubSetActive(false) });
+    const gitlabButton = document.getElementById("button2");
+    gitlabButton.addEventListener("mouseover", function gitlab () { gitlabSetActive(true) });
```

Comment:
This approach works, but it is not elegant and also not the react way of adding event listeners. You could do something as below

```js

const mouseOverFunction = () => {
    githubSetActive(true);
}

const mouseOutFunction = () => {
    githubSetActive(false);
}

<button style={githubActiveStyles} className="button" onMouseOver={mouseOverFunction} onMouseOut={mouseOutFunction}>
    <img src={githubActive ? activateGithub : github} alt="" />
    <span className="button-text">Continue with GitHub</span>
</button>
```

With this, you won't have to bother about getting element by id or removing the event listeners. React will automatically remove the event listeners when the component unmounts. You also won't need the `useEffect` hook.